bpfd/src/bpf.rs

```diff
@@ -992,9 +1013,25 @@ impl BpfManager {
 
         if let Some(map) = self.maps.get_mut(&index) {
             map.used_by.push(id);
+
+            // Update all the programs using the same map with the updated map_used_by.
+            for used_by_id in map.used_by.iter() {
+                if let Some(program) = self.programs.get_mut(used_by_id) {
+                    if let Ok(data) = program.data_mut() {
+                        data.set_maps_used_by(Some(map.used_by.clone()));
+                    }
```

Comment:
one liner ^^^

Reply:
clippy and fmt forced it back to what I had originally.